driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java

```diff
@@ -37,15 +38,15 @@
 import static com.mongodb.internal.connection.ProtocolHelper.sendCommandFailedEvent;
 import static com.mongodb.internal.connection.ProtocolHelper.sendCommandStartedEvent;
 import static com.mongodb.internal.connection.ProtocolHelper.sendCommandSucceededEvent;
-import static java.lang.String.format;
 
 class LoggingCommandEventSender implements CommandEventSender {
     private static final int MAX_COMMAND_DOCUMENT_LENGTH_TO_LOG = 1000;
+    private static final double NANOS_PER_MILLI = 1_000_000.0d;
 
     private final ConnectionDescription description;
     @Nullable private final CommandListener commandListener;
     private final RequestContext requestContext;
-    private final Logger logger;
+    private final StructuredLogger logger;
```

Comment:
Replaced Logger with StructuredLogger to allow implementation of the spec tests, which require the individual dynamic fields to be available for assertions.